app/models/transaction.rb

```diff
@@ -0,0 +1,6 @@
+class Transaction < ApplicationRecord
+  belongs_to :user
+
+  validates :amount, :date, presence: true
```

Comment:
Very simple validations here. What if the date in the future or negative amount for a transaction?

Reply:
I have tried to test this code:

validate :we_are_not_in_the_future

private

def we_are_not_in_the_future
  errors.add(:date, 'You can not set a date greater than current') if date > Time.zone.today
end
But got an error:

Failure/Error: errors.add(:date, 'You can not set a date greater than the current') if date > Time.now
     
     NoMethodError:
       undefined method `>' for nil:NilClass
I don't know why, because in the console I can create a new Transaction without any problem.

Reply:
@wazzuper if you create a new instance of the model with `Transaction.new` it won't have initialized `date` field at the moment. So you should check it:
```ruby
def we_are_not_in_the_future
  if date.present? && date > Time.zone.today
    errors.add(:date, 'You can not set a date greater than current') 
  end
end
```
I suggest you simplifying the name of the method to `date_in_the_past`:
```ruby
validate :date_in_the_past

private

def date_in_the_past
  ...
end
```

Reply:
Wow, thank you!